timesketch/lib/analyzers/interface.py

```diff
@@ -1150,3 +1165,196 @@ def get_kwargs(cls):
     def run(self):
         """Entry point for the analyzer."""
         raise NotImplementedError
+
+
+class AnalyzerOutputException(Exception):
+    """Analyzer output exception."""
+
+
+class AnalyzerOutput:
+    """A class to record timesketch analyzer output.
+
+    Attributes:
+        platform (str): [Required] Analyzer platfrom.
+        analyzer_identifier (str): [Required] Unique analyzer identifier.
+        analyzer_name (str): [Required] Analyzer display name.
+        result_status (str): [Required] Analyzer result status.
+            Valid values are success, no-findings, error.
+        dfiq_question_id (str): [Optional] Digital Forensics Investigative
+            Question (DFIQ) questions ID.
+        dfiq_question_conclusion (str): [Optional] DFQI questions answer/
+            conclusion.
+        result_priority (str): [Required] Priority of the result based on the
+            analysis findings. Valid values are NOTE (default), LOW, MEDIUM, HIGH.
+        result_summary (str): [Required] A summary statement of the analyzer
+            finding. A result summary must exist even if there is no finding.
+        result_markdown (str): [Optional] A detailed information about the
+            analyzer finding in a markdown format.
+        references (List[str]): [Optional] A list of references about the
+            analyzer or the issue the analyzer attempts to address.
+        attributes (dict): [Optional] A list of object that holds the
+            finding details.
+        timesketch_instance (str): [Required] The Timesketch instance URL.
+        sketch_id (int): [Required] The Timesketch sketch ID for this analyzer.
+        timeline_id (int): [Required] The Timesketch timeline ID for this analyzer.
+        saved_views (List[int]): [Optional] Saved views generatred by the analyzer.
+        saved_stories (List[int]): [Optional] Saved stories generated by the analyzer.
+        saved_graphs (List[int]): [Optional] Saved graphs generated by the analyzer.
+        saved_aggregations (List[int]): [Optional] Saved aggregations generated
+            by the analyzer.
+        created_tags (List[str]): [Optional] Tags created by the analyzer.
+    """
+
+    def __init__(
+        self,
+        analyzer_identifier,
+        analyzer_name,
+        timesketch_instance,
+        sketch_id,
+        timeline_id,
+    ):
+        """Initialize analyzer output."""
+        self.platform = "timesketch"
+        self.analyzer_identifier = analyzer_identifier
+        self.analyzer_name = analyzer_name
+        self.result_status = ""  # TODO: link to analyzer status/error?
+        self.dfiq_question_id = ""
+        self.dfiq_question_conclusion = ""
+        self.result_priority = "NOTE"
+        self.result_summary = ""
+        self.result_markdown = ""
+        self.references = []
+        self.attributes = {}
```

Comment:
The name `attributes` is too generic IMO. Can we rename this to something more telling, e.g. `result_attributes`?

Reply:
Yes, that should work as well.
```suggestion
        self.result_attributes = {}
```